### DISCUSSION TITLE
[Question][HW1] Implicit conversion for arithmetic ops & unsigned compare ops

### DISCUSSION THREAD

**Discussion by retroinspect** [prosyslab-classroom/cs424-program-reasoning#381] [Q&A]
1. Is it an undefined behavior to put boolean operands for arithmetic/comparison operators? The doc(homework.pdf) says 
> **The semantics of ⊕ and < follows the standard definition of arithmetic and comparison operators of LLVM**

but I cannot find the specification of implicit conversion nor explicit error from [LLVM Language Reference](https://llvm.org/docs/LangRef.html).

From what I understood, LLVM IR is motivated from assembly, so it sounds plausible if we support implicit conversion such as
`add %rax, %rax, %ebx`

2. Why do unsigned comparison operators(e.g. Ugt, Uge, Ult, Ule) exist in `Llvm.Icmp` although doc(homework.pdf) says 
> However, the interpreter will not take into account the sign of operators. 

3. Are there extra test cases for grading?

**Comment by KihongHeo** [comment]
Hi.

1. Ignore implicit conversion in this homework.
2. Ignore unsigned comparisons in this homework. In the homework, we will only consider a subset of the full LLVM operators (e.g., `Llvm.Icmp`).
3. Yes.
